src/xGadgetron/cGadgetron/gadgetron_data_containers.h

```diff
@@ -248,6 +260,10 @@ namespace sirf {
 			std::cerr << "AcquisitionsFile::set_acquisition not implemented yet, sorry\n";
 		}
 		virtual void append_acquisition(ISMRMRD::Acquisition& acq);
+
+		virtual void append_sptr_acquisition( gadgetron::shared_ptr<ISMRMRD::Acquisition> sptr_acqu){throw LocalisedException("Function undefined for sirf::AcuisitionsFile.", __FILE__, __LINE__);};
+		virtual gadgetron::shared_ptr<ISMRMRD::Acquisition> get_sptr_acquisition(unsigned int num){throw LocalisedException("Function undefined for sirf::AcuisitionsFile.", __FILE__, __LINE__);};
```

Comment:
as above `get_acquisition_sptr`. should also be a `const` member.

But do we really need this? Providing access to internal `shared_ptr` can be quite dangerous.

Reply:
I suggested this change becaues in the MR side of the simulation framework, arbitrarily many dynamics are enabled, e.g. one could add two motion and contrast variations for four different tissue types. Then I keep track of which readout line is acquired in which dynamic state and simulate the data which the `ISMRMRD::Acquisition` carries in the respective dynamic state.

So all `ISMRMRD::Acquisition` readout lines from some template file are read into memory, but I need to keep track of them with a pointer, otherwise the memory use explodes. 

We want to use files of size ~5GB, flexibly duplicating them during the simulation blows up the memory (of my laptop).

Reply:
Is there some other way to have objects just exist once and manipulate them in a safer way?

Reply:
I see you have an example at #243. I'll add a question there soon.

Reply:
Making the `get_acquisition_sptr()` a `const` function is non-trivial unfortunately.
This requires `items()`and `number()`to be `const` as well and these methods come from the interface `aDataContainer` so this is a larger-scale operation I don't want to get into in this PR, haha, also since there is the new data hierarchy coming from @evgueni-ovtchinnikov I think which might involve changes there.